### PR TITLE
Using docker Kafka container for testing

### PR DESCRIPTION
**Pull request by kokosing**
Using docker Kafka container for testing

### CODE REVIEW

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaDistributed.java [hunk 1]
```diff
@@ -14,38 +14,37 @@
...
 
     public TestKafkaDistributed()
             throws Exception
     {
-        this(createEmbeddedKafka());
+        this(new TestingKafka());
     }
 
-    public TestKafkaDistributed(EmbeddedKafka embeddedKafka)
+    public TestKafkaDistributed(TestingKafka testingKafka)
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaDistributed.java]: Use this opportunity to convert this into `getQueryRunner()` method

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import static org.testcontainers.containers.KafkaContainer.KAFKA_PORT;
+
+public class TestingKafka
+        implements Closeable
+{
+    private final KafkaContainer kafkaContainer;
+
+    public TestingKafka()
+    {
+        kafkaContainer = new KafkaContainer("5.2.1");
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java]: am i correct we were using 1.1.1 for testing before?
please upgrade version and refactor tests separately
or, if not possible, reflect this in cmt msg

like in https://github.com/prestosql/presto/pull/2163#discussion_r352634164

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    {
+        public CloseableProducer(ProducerConfig config)
+        {
+            super(config);
+        }
+    }
+
+    public String getConnectString()
+    {
+        return "localhost:" + kafkaContainer.getMappedPort(KAFKA_PORT);
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java]: `kafkaContainer.getcontainerip() + ":" ... `

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            extends kafka.javaapi.producer.Producer<K, V>
+            implements AutoCloseable
+    {
+        public CloseableProducer(ProducerConfig config)
+        {
+            super(config);
+        }
+    }
+
+    public String getConnectString()
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java]: move above `public static class CloseableProdu`

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java (hunk 2 above)
Comment (kokosing) [reply presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java]: This is a bit more complicated. See https://docs.confluent.io/current/installation/versions-interoperability.html

Docker version is a version of confluent. So it looks like we were using 4.1.x kind of confluent version.

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaDistributed.java (hunk 1 above)
Comment (kokosing) [reply presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaDistributed.java]: I guess you meant `createQueryRunner`

**Pull request merged**